### ISSUE TITLE
Permalinks to tasks categorized under task lists in BuddyPress don't work

### ISSUE THREAD

**Issue by ericandrewlewis** [WebDevStudios/CollabPress#43]

**Comment by boonebgorges** [comment]
This looks like it's probably related to the URL schema change.

One general strategy (which will handle this problem as well as other permalink-related ones) is to filter `bp_get_canonical_url`. In that filter, do something like:

```
if ( this URL looks like it should be a collabpress link ) {
    if ( URL looks like it contains a task-list ) {
        get the name of the project from the last part of the URL;
        $canonical_url = the permalink of that project;
    }
}
return $canonical_url;
```

**Comment by ericandrewlewis** [comment]
Fixed in f03dee957d6dd4363cfaa67dc4c55569f053189a
